src/main/java/org/jolokia/docker/maven/assembly/DockerAssemblyManager.java

```diff
@@ -262,8 +263,13 @@ DockerFileBuilder createDockerFileBuilder(BuildImageConfiguration buildConfig, A
             builder.workdir(buildConfig.getWorkdir());
         }
         if (assemblyConfig != null) {
+            String baseDir = assemblyConfig.getBasedir();
+            if (assembly != null && assembly.getBaseDirectory() != null) {
+                baseDir = assembly.getBaseDirectory();
+            }
```

Comment:
The problem here is, if you set the *outer* basedir (the directory, that is used by d-m-p to place the assembly within a docker image) to the same as the *inner* basedir of the assembly descriptor you will get the directories, twice (e.g. if you set the assembly `baseDirectory` to `/app-dir` then your files will be placed into `/app-dir/app-dir`. That's because the `AssemblyArchiver` uses this directory, too create an extra level. 

In fact in this case the outer `baseDir` should be set to `/` to avoid this.